components/AddExercises/ExerciseList.jsx

```diff
@@ -36,92 +39,111 @@ const Buttontext = styled.Text`
+  };
+
+  const handleSortByMuscleGroupPress = () => {
+    setDataState({ isSortByMuscleGroup: true, filteredDataSource: parsedItemsMuscleGroups, masterDataSource: parsedItemsMuscleGroups });
+  };
 
   const searchFilterFunction = (text) => {
-    // Check if searched text is not blank
     if (text) {
-      // Inserted text is not blank
-      // Filter the masterDataSource
-      // Update FilteredDataSource
-      const firstLetter = text[0].toUpperCase();
-      if (masterDataSource[firstLetter]) {
-        const newData = masterDataSource[firstLetter].filter((item) => {
+      const filteredItems = [];
+      const textData = text.toUpperCase();
+      Object.keys(dataState.masterDataSource).forEach((key) => {
+        const newData = dataState.masterDataSource[key].filter((item) => {
           const itemData = item.name
             ? item.name.toUpperCase()
             : '';
-          const textData = text.toUpperCase();
           return itemData.indexOf(textData) > -1;
         });
-        const filteredData = {};
-        filteredData[firstLetter] = newData;
-        setFilteredDataSource(filteredData);
-        setSearch(text);
+        if (newData) {
+          Object.keys(newData).forEach((each) => {
+            if (newData[each]) {
+              filteredItems.push(newData[each]);
+            }
+          });
+        }
+        filteredItems.sort((a, b) => a.name.localeCompare(b.name));
+      });
+      setDataState({ ...dataState, filteredDataSource: filteredItems });
+      setSearch(text);
+    } else {
+      setDataState({ ...dataState, filteredDataSource: dataState.masterDataSource });
+      setSearch(text);
+    }
+  };
+
+  const handleUnfilteredCardPress = (item, index) => {
+    const allExercises = { ...dataState.masterDataSource };
+    if (dataState.isSortByMuscleGroup) {
+      const { selected } = allExercises[item.muscleGroups][index];
+      allExercises[item.muscleGroups][index].selected = !selected;
+      if (allExercises[item.muscleGroups][index].selected === true) {
+        setExerciseCount(exerciseCount + 1);
+        addedExercises.push(item);
       } else {
-        setFilteredDataSource({});
-        setSearch(text);
+        setExerciseCount(exerciseCount - 1);
+        addedExercises.splice(addedExercises.indexOf(item), 1);
       }
     } else {
-      // Inserted text is blank
-      // Update FilteredDataSource with masterDataSource
-      setFilteredDataSource(masterDataSource);
-      setSearch(text);
+      const { selected } = allExercises[item.name[0]][index];
+      allExercises[item.name[0]][index].selected = !selected;
+      if (allExercises[item.name[0]][index].selected === true) {
+        setExerciseCount(exerciseCount + 1);
+        addedExercises.push(item);
+      } else {
+        setExerciseCount(exerciseCount - 1);
+        addedExercises.splice(addedExercises.indexOf(item), 1);
+      }
     }
+    setDataState({ ...dataState, masterDataSource: allExercises });
+  };
```

Comment:
I think we can get rid of the if/else block here. If we just conditionally save the first array key in a variable (item.muscleGroups or item.name[0]) you can reuse all the code inside the if/else block.